Apps/W1/EDocumentConnectors/Tietoevry/app/src/ConnectionSetupCard.Page.al

```diff
@@ -0,0 +1,138 @@
+                    ToolTip = 'Specifies the client ID.';
+                    ApplicationArea = Basic, Suite;
+                    ExtendedDatatype = Masked;
+                    ShowMandatory = true;
+
+                    trigger OnValidate()
+                    begin
+                        this.TietoevryAuth.SetClientId(Rec."Client ID - Key", this.ClientID);
+                    end;
+                }
+                field(ClientSecret; this.ClientSecret)
+                {
+                    Caption = 'Client Secret';
+                    ToolTip = 'Specifies the client secret.';
+                    ApplicationArea = Basic, Suite;
+                    ExtendedDatatype = Masked;
+                    ShowMandatory = true;
+
+                    trigger OnValidate()
+                    begin
+                        this.TietoevryAuth.SetClientSecret(Rec."Client Secret - Key", this.ClientSecret);
+                    end;
+                }
+                field("Authentication URL"; Rec."Authentication URL")
+                {
+                    ApplicationArea = Basic, Suite;
+                    ToolTip = 'Specifies the URL to connect to Avalara.';
+                }
+                field("API URL"; Rec."API URL")
+                {
+                    ApplicationArea = Basic, Suite;
+                    ToolTip = 'Specifies the URL to connect to Avalara''s api.';
+                }
+                field("Sandbox Authentication URL"; Rec."Sandbox Authentication URL")
+                {
+                    ApplicationArea = Basic, Suite;
+                    ToolTip = 'Specifies the URL to connect to Avalara sandbox.';
+                }
+                field("Sandbox API URL"; Rec."Sandbox API URL")
+                {
+                    ApplicationArea = Basic, Suite;
+                    ToolTip = 'Specifies the URL to connect to Avalara sandbox api.';
+                }
+                field("Company Id"; Rec."Company Id")
+                {
+                    ApplicationArea = Basic, Suite;
+                    ToolTip = 'Specifies the company ID.';
+                }
+                field("Send Mode"; Rec."Send Mode")
+                {
+                    ApplicationArea = Basic, Suite;
+                    ToolTip = 'Specifies the send mode.';
+                    ShowMandatory = true;
+                }
+            }
+        }
+    }
+    actions
+    {
+        area(processing)
+        {
+            action(Authenticate)
+            {
+                ApplicationArea = Basic, Suite;
+                Caption = 'Authenticate';
+                Image = Setup;
+                Promoted = true;
+                PromotedCategory = Process;
+                PromotedOnly = true;
+                ToolTip = 'Verify the Authentication Details';
+
+                trigger OnAction()
+                var
+                    TietoevryAuth: Codeunit Authenticator;
+                    [NonDebuggable]
```

Comment:
No need. It is secret text

Reply:
No, the variables are Text and used locally in the page to pass information to TietoevryAuth.